### ISSUE TITLE
How to debug

### ISSUE THREAD

**Issue by blxl909** [PUTvision/qgis-plugin-deepness#91]
Hello, I'm enjoying using your plugin, and I want to modify this code to fit my own project requirements.
I have tried debugvs and First Aid plugin, but maybe because QGSTask running in background, I can't hit the break point I set in code such as segmentor.py (my currently solution is to raise exception to force plugin stop to debug, but I think it's too complicated )
I'm using visual studio code as IDE. do you have any suggestion? Thanks for your help.

**Comment by przemyslaw-aszkowski** [comment]
Hello, 
Nice to hear your positive feedback. 
While developing the code, in order to debug, we used unit tests, e.g. `test_map_processor_segmentation.py`. It employs the same code and can be run without QGIS. I can definitively recommend this method. 
We did not debug inside QGIS, though I suspect it would require attaching the debugger to some specific process.
Let me know in case of issues, and feel invited to share your ideas and application in [User stories](https://github.com/PUTvision/qgis-plugin-deepness/discussions/86) :)
Cheers
Przemek
